integrations/google_ai/src/haystack_integrations/components/embedders/google_ai/text_embedder.py

```diff
@@ -0,0 +1,205 @@
+        """
+        Deserializes the component from a dictionary.
+
+        :param data:
+            Dictionary to deserialize from.
+        :returns:
+            Deserialized component.
+        """
+        # Ensure api_key is properly deserialized from Secret
+        deserialize_secrets_inplace(data["init_parameters"], keys=["api_key"])
+        return default_from_dict(cls, data)
+
+    @component.output_types(embedding=List[List[float]], meta=Dict[str, Any])
+    def run(self, texts: List[str]):
+        """
+        Embeds a list of texts using the configured Google AI model.
+
+        :param texts: A list of strings to embed.
+        :returns: A dictionary containing:
+            - `embedding`: A list of embeddings, where each embedding is a list of floats.
+            - `meta`: A dictionary with metadata about the operation (e.g., model name, task type).
+        :raises TypeError: If the input `texts` is not a list of strings.
+        :raises RuntimeError: If the embedding process fails.
+        """
+        if not isinstance(texts, list) or not all(isinstance(text, str) for text in texts):
+            msg = "GoogleAIGeminiTextEmbedder expects a List of strings as input."
+            raise TypeError(msg)
+        if not texts:
+            # Return empty list if no texts are provided
+            return {"embedding": [], "meta": {"model": self.model, "task_type": self.task_type}}
+
+        if self._api_key_resolved is None:
+            msg = "The component has not been warmed up. Please call warm_up() before running."
+            raise RuntimeError(msg)
+
+        # Prepare parameters for the API call
+        configs = types.EmbedContentConfig()
+        api_params = {"model": self.model, "contents": texts, "configs": configs}
+
+        if self.task_type:
+            configs.task_type = self.task_type
+        # Add title only if task_type is retrieval_document and title is provided
+        if self.task_type == "retrieval_document" and self.title:
+            configs.title = self.title
+        elif self.title and self.task_type != "retrieval_document":
+            warnings.warn(
+                UserWarning("Warning: Title 'Should Be Ignored' is ignored because task_type is 'retrieval_query'"),
+                stacklevel=2,
+            )
+        if self.output_dimensionality:
+            configs.output_dimensionality = self.output_dimensionality
+        try:
+            # Make the API call to embed the batch of texts
+            result = self.client.models.embed_content(**api_params)
+
+        except Exception as e:
+            # TODO: Add more specific error handling for common API errors if possible
+            msg = f"Google AI embedding failed: {e}"
+            raise RuntimeError(msg) from e
+
+        # Extract embeddings - result.embedding should be the list of lists
+        embeddings = result.get("embedding")  # Use .get for safety, returns None if key missing
```

Comment:
According to [docs](https://ai.google.dev/api/embeddings), result should be an object and not a `dict`, so you should do `result.embeddings`.

I see that in the tests you're mocking this response (so tests are actually passing), but have you tried it outside tests (e.g. in an integration tests or an example?)